channels/auth.py

```diff
@@ -20,7 +22,7 @@
 
 
 @database_sync_to_async
-def get_user(scope):
+def get_user(scope: Dict[str, Any]) -> Awaitable[Union[User, AnonymousUser]]:
```

Comment:
would it not make more sense to make the `database_sync_to_async` be generic

```python
def database_sync_to_async(method: Callable[SomeGenericVar...]) -> Callable[ Awaitable[TheGenericReturnOfTHeCallable]]
....
```

Reply:
It's not that easy, database_sync_to_async inherits from SyncToAsync class from asgiref library.
Are you suggesting that I should inherit the __call__ method to then type it like you wrote?

Reply:
good point, no lets just leave that as is.